processor/src/main/resources/org/mapstruct/ap/internal/model/MethodReference.ftl

```diff
@@ -20,7 +20,16 @@
 -->
 <@compress single_line=true>
     <#-- method is either internal to the mapper class, or external (via uses) declaringMapper!=null -->
-    <#if declaringMapper??><#if static><@includeModel object=declaringMapper.type/><#else>${mapperVariableName}</#if>.</#if>${name}<#if (parameters?size > 0)>( <@arguments/> )<#else>()</#if>
+    <#if declaringMapper??><#if static><@includeModel object=declaringMapper.type/><#else>${mapperVariableName}</#if>.<@params/>
+    <#elseif static && mapperToImplement??><@includeModel object=mapperToImplement/>.<@params/>
```

Comment:
Good question.. I'll add a test case for that.. 

Which brings me to another point.. I've been struggling with the integration test (finally managed, but handy is not the term I would use).. Isn't it time to add java 8 test to the standard unit test somehow (facilitating more easy debug, and less risk)..